go/mysql/connection.go

```diff
@@ -47,5 +48,11 @@ func (this *ConnectionConfig) Equals(other *ConnectionConfig) bool {
 }
 
 func (this *ConnectionConfig) GetDBUri(databaseName string) string {
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", this.User, this.Password, this.Key.Hostname, this.Key.Port, databaseName)
+	var ip = net.ParseIP(this.Key.Hostname)
+	if (ip != nil) && (ip.To4() == nil) {
```

Comment:
To answer your original question, the Go standard library will pack an IPv4 address (with zeros) into a 16-byte slice if you call .To16 on it, so it's not a useful check.

It frustrates me immensely that the language doesn't ship with a `.AddressFamily()` discriminator or a `.IsIpv4()` function or a way to call the `inet_pton` function.

For your second question, it seems stylistically preferable to me to check `ip != nil` because that does something conceptually different – it checks whether the passed input is an IP address literal at all. I'm of the school that explicit is always better than implicit.